src/ore/src/id_gen.rs

```diff
@@ -100,19 +101,21 @@ pub struct IdAllocatorInnerBitSet {
     next: StdRng,
     min: u32,
     max: u32,
+    mask: u32,
     used: BitSet,
 }
 
 impl IdAllocatorInner for IdAllocatorInnerBitSet {
     const NAME: &'static str = "hibitset";
 
-    fn new(min: u32, max: u32) -> Self {
+    fn new(min: u32, max: u32, mask: u32) -> Self {
```

Comment:
If we wanted to add an assert here, I think something like the following would work:

```
let mask_check = (1 << log_2(max)) - 1;
assert_eq!(mask & mask_check, 0);
```

Reply:
Neat, thanks!